### ISSUE TITLE
[Component] [dialog] dialog中使用TinyMCE富文本编辑器,弹框中input无法聚焦

### ISSUE THREAD

**Issue by 270556660** [element-plus/element-plus#9347]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.37`
- Element Plus Version: `2.2.13`
- Browser / OS: `Chrome 103.0.5060.66 / win10`
- Build Tool: `Vite`

### Related Component

- `el-dialog`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgRWxEaWFsb2cgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBkaWFsb2dWaXNpYmxlID0gcmVmKHRydWUpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZWwtZGlhbG9nIHYtbW9kZWw9XCJkaWFsb2dWaXNpYmxlXCIgYXBwZW5kLXRvLWJvZHk+XG4gICAgZGlhbG9n5by556qX5L2/55SoVGlueU1DReWvjOaWh+acrOe8lui+keWZqCzlvLnmoYbkuK3nmoRpbnB1dOaXoOazleiBmueEplxuICA8L2VsLWRpYWxvZz5cbiAgPGRpdiBjbGFzcz1cInBvcFwiPlxuICAgIDxpbnB1dCB0eXBlPVwidGV4dFwiIGNsYXNzPVwiaW5wXCIgcGxhY2Vob2xkZXI9XCLov5nph4zml6Dms5XogZrnhKZcIi8+XG4gIDwvZGl2PlxuPC90ZW1wbGF0ZT5cbjxzdHlsZT5cbiAgLnBvcHtcbiAgICBwb3NpdGlvbjogZml4ZWQ7IFxuICAgIHotaW5kZXg6IDMwMDA7XG4gIH1cbiAgLmlucHtcbiAgICBib3JkZXI6c29saWQgMXB4ICMzMzM7XG4gIH1cbjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

```vue
<script setup lang="ts">
import { ref, version as vueVersion } from 'vue'
import { ElDialog } from 'element-plus'
import { ElementPlus } from '@element-plus/icons-vue'

const dialogVisible = ref(true)
</script>

<template>
  <el-dialog v-model="dialogVisible" append-to-body>
    dialog弹窗使用TinyMCE富文本编辑器,弹框中的input无法聚焦
  </el-dialog>
  <div class="pop">
    <input type="text" class="inp" placeholder="这里无法聚焦"/>
  </div>
</template>
<style>
  .pop{
    position: fixed; 
    z-index: 3000;
  }
  .inp{
    border:solid 1px #333;
  }
</style>
```

### What is Expected?

使用dialog,附加到body窗口中的input聚焦
    

### What is actually happening?

由于TinyMCE将它们的对话框附加到body而不是dialog窗口，因此它们被认为是在dialog之外，并且无法聚焦。

### Additional comments

_(empty)_

![image](https://user-images.githubusercontent.com/13551159/185031841-685f4a7b-d495-433d-a3a7-ae26fd8ee458.png)

**Comment by btea** [comment]
可以将`append-to-body` 设置为 `false`

**Comment by Felix19901229** [comment]
这个不是element的bug，这个应该是层级问题（z-index）,建议将TinyMCE的层级设置成3000以上（element z-index在2000以上）。还有一个比较隐藏的bug，就是页面如果开启自动翻译，基本上富文本编辑器的光标都会有一些莫名其妙的问题。（被TinyMCE坑怕了，这个东西不好用，又太复杂，还是国产wangeditor好使，只不过功能性差一点）最好在页面上设置禁止翻译属性。

**Comment by 270556660** [comment]
> 这个不是element的bug，这个应该是层级问题（z-index）,建议将TinyMCE的层级设置成3000以上（element z-index在2000以上）。还有一个比较隐藏的bug，就是页面如果开启自动翻译，基本上富文本编辑器的光标都会有一些莫名其妙的问题。（被TinyMCE坑怕了，这个东西不好用，又太复杂，还是国产wangeditor好使，只不过功能性差一点）最好在页面上设置禁止翻译属性。

不是z-index问题，通过监听focusin事件，焦点被锁定在.el-dialog框内

**Comment by 270556660** [comment]
> 可以将`append-to-body` 设置为 `false`

v2.2.0之前版本不需要append-to-body

**Comment by btea** [comment]
`append-to-body` 默认值应该是 `false`, 你不设置就行。

**Comment by wjr1116** [comment]
请问解决了吗？不止是tinymce，只要是el-dialog或el-drawer里面还有二次弹窗，那个弹窗里的输入框都是无法聚焦的

**Comment by btea** [comment]
> 请问解决了吗？不止是tinymce，只要是el-dialog或el-drawer里面还有二次弹窗，那个弹窗里的输入框都是无法聚焦的

能提供一个例子吗？

**Comment by wjr1116** [comment]
> > 请问解决了吗？不止是tinymce，只要是el-dialog或el-drawer里面还有二次弹窗，那个弹窗里的输入框都是无法聚焦的
> 
> 能提供一个例子吗？

比如上面说的dialog里的tinymce的弹窗，还有像在dialog里用<el-date-picker  type="datetime" ... />
![image](https://user-images.githubusercontent.com/32118441/188684652-98a2e6b8-0258-4ad9-8b7c-0f199689f7c7.png)
上面的时间选择也没法聚焦

**Comment by btea** [comment]
@wjr1116  这样似乎可以，[链接](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgRWxEaWFsb2cgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBkaWFsb2dWaXNpYmxlID0gcmVmKHRydWUpXG5jb25zdCB2YWx1ZSA9IHJlZignJylcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxlbC1kaWFsb2cgdi1tb2RlbD1cImRpYWxvZ1Zpc2libGVcIiA+XG4gICAgZGlhbG9n5by556qX5L2/55SoVGlueU1DReWvjOaWh+acrOe8lui+keWZqCzlvLnmoYbkuK3nmoRpbnB1dOaXoOazleiBmueEplxuICAgIDxlbC1kYXRlLXBpY2tlciB2LW1vZGVsPVwidmFsdWVcIiB0eXBlPVwiZGF0ZXRpbWVcIj48L2VsLWRhdGUtcGlja2VyPlxuICA8L2VsLWRpYWxvZz5cbiAgXG48L3RlbXBsYXRlPlxuPHN0eWxlPlxuICAucG9we1xuICAgIHBvc2l0aW9uOiBmaXhlZDsgXG4gICAgei1pbmRleDogMzAwMDtcbiAgfVxuICAuaW5we1xuICAgIGJvcmRlcjpzb2xpZCAxcHggIzMzMztcbiAgfVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

**Comment by wjr1116** [comment]
@btea 好吧，原来是我element-plus 2.2.2的版本会有这个bug

**Comment by wjr1116** [comment]
@btea 不过dialog的tinymce富文本点击工具栏上的弹窗的输入框还是没法聚焦

**Comment by btea** [comment]
tinymce富文本在 `dialog` 里的吗？

**Comment by wjr1116** [comment]
> tinymce富文本在 `dialog` 里的吗？

是的

**Comment by btea** [comment]
能在[stackblitz.](https://stackblitz.com/)或者[codesandbox](https://codesandbox.io/)创建一个复现的仓库吗？

**Comment by youcannot** [comment]
只能这么干了

**Comment by youcannot** [comment]
这样可以，防在dialog内部  <el-dialog v-model="dialogVisible" append-to-body>
    dialog弹窗使用TinyMCE富文本编辑器,弹框中的input无法聚焦
    <div class="pop">
        <input type="text" class="inp" placeholder="这里无法聚焦"/>
      </div>
  </el-dialog>

**Comment by youcannot** [comment]
今天我也遇到了，只能将其放在 dialog 内部 才能正确获取焦点

**Comment by btea** [comment]
fixed by #10160

**Comment by tgv587** [comment]
![image](https://user-images.githubusercontent.com/56184848/215307397-869366fc-8157-4426-9106-489a61d7d8ac.png)
默认弹框内部会监听focusin焦点失去事件，当焦点失去时会判断当前触发事件对象dom是否在弹框内部，如果不在内部则会触发如上图的tryFocus方法，它会强行把焦点再次聚集到上一次的焦点dom对象中，简单的处理办法在node_modules\element-plus\es\components\focus-trap\src\utils2.mjs找到对应源码注释掉element.focus({ preventScroll: true })即可。

**Comment by 1215708470** [comment]
> ![image](https://user-images.githubusercontent.com/56184848/215307397-869366fc-8157-4426-9106-489a61d7d8ac.png) 默认弹框内部会监听focusin焦点失去事件，当焦点失去时会判断当前触发事件对象dom是否在弹框内部，如果不在内部则会触发如上图的tryFocus方法，它会强行把焦点再次聚集到上一次的焦点dom对象中，简单的处理办法在node_modules\element-plus\es\components\focus-trap\src\utils2.mjs找到对应源码注释掉element.focus({ preventScroll: true })即可。

你这个方法对于drawer组件并没有作用
